apps/settings/js/panels/call/call.js

```diff
@@ -314,146 +309,86 @@ require([
-      var onerror = function call_getCWOptionError() {
-        if (callback) {
-          _ignoreSettingChanges = false;
-          callback(null);
-        }
-      };
-
-      // Queries rules for unconditional call forwarding.
-      var unconditional = _mobileConnection.getCallForwardingOption(
-        CallConstant.CALL_FORWARD_REASON.UNCONDITIONAL);
-
-      unconditional.onsuccess = function() {
-        var unconditionalRules = unconditional.result;
-
-        // Queries rules for call forwarding when device busy.
-        var mobileBusy = _mobileConnection.getCallForwardingOption(
-          CallConstant.CALL_FORWARD_REASON.MOBILE_BUSY);
-
-        mobileBusy.onsuccess = function() {
-          var mobileBusyRules = mobileBusy.result;
-
-          // Queries rules for call forwarding when device does not reply.
-          var noReply = _mobileConnection.getCallForwardingOption(
-            CallConstant.CALL_FORWARD_REASON.NO_REPLY);
-
-          noReply.onsuccess = function() {
-            var noReplyRules = noReply.result;
-
-            // Queries rules for call forwarding when device is not reachable.
-            var notReachable = _mobileConnection.getCallForwardingOption(
-              CallConstant.CALL_FORWARD_REASON.NOT_REACHABLE);
-
-            notReachable.onsuccess = function() {
-              var notReachableRules = notReachable.result;
-
-              var cfOptions = {
-                'unconditional': unconditionalRules,
-                'mobilebusy': mobileBusyRules,
-                'noreply': noReplyRules,
-                'notreachable': notReachableRules
-              };
+      var cardIndex = DsdsSettings.getIccCardIndexForCallSettings();
+
+      CallForwarding.getAllOptions(cardIndex).then(function onSuccess(results) {
+        var cfOptions = {
+          'unconditional': results[0],
+          'mobilebusy': results[1],
+          'noreply': results[2],
+          'notreachable': results[3]
+        };
 
-              // Waits for all DB settings completed.
-              var asyncOpChecker = {
-                taskCount: 0,
-                runTask: function(func) {
-                  this.taskCount++;
-                  var newArgs = [];
-                  for (var i = 1; i < arguments.length; i++) {
-                    newArgs.push(arguments[i]);
-                  }
-                  newArgs.push(this.complete.bind(this));
-                  func.apply(window, newArgs);
-                },
-                complete: function() {
-                  this.taskCount--;
-                  if (this.taskCount === 0) {
-                    this.finish();
-                  }
-                },
-                finish: function() {
-                  setTimeout(function() {
-                    _ignoreSettingChanges = false;
-                    callback(cfOptions);
-                  }, 500);
-                }
-              };
+        AsyncOperator.finishCallback = function() {
```

Comment:
We didn't have promise before. Now we can simply use `Promise.all` and this module should go.